examples/asr/speech2text.py

```diff
@@ -0,0 +1,175 @@
+)
+parser.add_argument(
+    "--amp_opt_level",
+    default="O0",
+    type=str,
+    choices=["O0", "O1", "O2", "O3"],
+    help="See: https://nvidia.github.io/apex/amp.html",
+)
+parser.add_argument("--train_dataset", type=str, required=True, default=None, help="training dataset path")
+parser.add_argument("--eval_datasets", type=str, nargs="*", help="evaluation datasets paths")
+parser.add_argument("--eval_freq", default=1000, type=int, help="Evaluation frequency")
+parser.add_argument("--eval_batch_size", type=int, default=8, help="batch size to use for evaluation")
+parser.add_argument("--local_rank", default=None, type=int, help="node rank for distributed training")
+parser.add_argument("--stats_freq", default=25, type=int, help="frequency with which to update train stats")
+parser.add_argument("--checkpoint_dir", default=None, type=str, help="Folder where to save checkpoints")
+parser.add_argument("--checkpoint_save_freq", required=False, type=int, help="how often (steps) to save checkpoints")
+parser.add_argument("--optimizer", default="novograd", type=str)
+parser.add_argument("--warmup_ratio", default=0.02, type=float, help="learning rate warmup ratio")
+parser.add_argument("--batch_size", required=True, type=int, help="train batch size per GPU")
+parser.add_argument("--num_epochs", default=5, type=int, help="number of epochs to train")
+parser.add_argument("--lr", default=0.01, type=float)
+parser.add_argument("--beta1", default=0.95, type=float)
+parser.add_argument("--beta2", default=0.5, type=float)
+parser.add_argument("--weight_decay", default=0.001, type=float)
+parser.add_argument("--iter_per_step", default=1, type=int, help="number of grad accumulations per batch")
+parser.add_argument("--wandb_exp_name", default=None, type=str)
+parser.add_argument("--wandb_project", default=None, type=str)
+args = parser.parse_args()
+logging = nemo.logging
+
+
+def main():
+    # Setup NeuralModuleFactory to control training
+    # instantiate Neural Factory with supported backend
+    nf = nemo.core.NeuralModuleFactory(
+        placement=nemo.core.DeviceType.CPU,
+        local_rank=args.local_rank,  # This is necessary for distributed training
+        optimization_level=args.amp_opt_level,  # This is necessary for mixed precision optimization
+        cudnn_benchmark=True,
+    )
+
+    # Instantiate the model which we'll train
+    if args.asr_model.endswith('.yaml'):
+        logging.info(f"Speech2Text: Will train from scratch using config from {args.asr_model}")
+        asr_model = nemo_asr.models.ASRConvCTCModel.import_from_config(args.asr_model)
+    else:
+        logging.info(f"Speech2Text: Will fine-tune from {args.asr_model}")
+        asr_model = nemo_asr.models.ASRConvCTCModel.from_pretrained(
+            model_info=args.asr_model, local_rank=args.local_rank
+        )
```

Comment:
You should consider spliting this argument into two

Reply:
but then I would need to add a check that only one is specified. Let's keep it like this.